data/store/scripts/templeteleport.lua

```diff
@@ -0,0 +1,23 @@
+local MIN_DISTANCE = 0
+
+function onRender(player, offer)
+	if player:isPzLocked() then
+		return false, "You can not purchase a transportation service while PZ Locked."
+	end
+
+	if Tile(player:getPosition()):hasFlag(TILESTATE_NOLOGOUT) then
+		return false, "You can not purchase a transportation service while in a no-logout zone."
+	end
+
+	local templePosition = player:getTown():getTemplePosition()
+	if player:getPosition():getDistance(templePosition) < MIN_DISTANCE then
+		return false, "You can not purchase this transportation service this close to your temple."
+	end
+
+	return true
+end
+
+function onBuy(player, offer)
+	player:teleportTo(templePosition)
```

Comment:
Hi, I was testing the system and it gives me an error on this line when purchasing a temple teleport. Console error is here:

`Lua Script Error: [Store Interface] 
data/store/scripts/templeteleport.lua:onBuy
attempt to index a nil value
stack traceback:
	[C]: at 0x010851cea0
	[C]: in function 'teleportTo'
	data/store/scripts/templeteleport.lua:21: in function <data/store/scripts/templeteleport.lua:20>`